util/prep-bazel-airgapped-build.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+set -euo pipefail
+
+: "${REPO_TOP:=$(git rev-parse --show-toplevel)}"
+
+: "${BAZELISK:=$REPO_TOP/.bin/bazelisk}"
+
+: "${BAZEL_AIRGAPPED_DIR:=bazel-airgapped}"
+: "${BAZEL_DISTDIR:=bazel-distdir}"
+: "${BAZEL_CACHEDIR:=bazel-cache}"
+
+LINE_SEP="====================================================================="
+
+################################################################################
+# Check if a previous airgapped directory has been built.
+################################################################################
+if [[ -d ${BAZEL_AIRGAPPED_DIR} ]]; then
+  if [[ $# == 0 || $1 != "--force-rebuild" ]]; then
+    while true; do
+      read -p "Airgapped directory exists, rebuild? [Y/n]" yn
+      case $yn in
+          "") rm -rf ${BAZEL_AIRGAPPED_DIR}; break;;
+          [Yy]*) rm -rf ${BAZEL_AIRGAPPED_DIR}; break;;
+          [Nn]*) exit;;
+          *) echo "Please enter [Yy] or [Nn]."
+      esac
+    done
+  fi
+fi
+
+################################################################################
+# Setup the airgapped directories.
+################################################################################
+mkdir -p ${BAZEL_AIRGAPPED_DIR}
+mkdir -p ${BAZEL_AIRGAPPED_DIR}/${BAZEL_DISTDIR}
+mkdir -p ${BAZEL_AIRGAPPED_DIR}/${BAZEL_CACHEDIR}
+cd ${BAZEL_AIRGAPPED_DIR}
+
+################################################################################
+# Prepare the distdir.
+################################################################################
+echo $LINE_SEP
+echo "Preparing bazel offline distdir ..."
+git clone https://github.com/bazelbuild/bazel
+cd bazel
+git checkout "$(cat ${REPO_TOP}/.bazelversion)"
+${BAZELISK} build @additional_distfiles//:archives.tar
+tar xvf bazel-bin/external/additional_distfiles/archives.tar \
+  -C "../${BAZEL_DISTDIR}" \
+  --strip-components=3
+cd ..
+rm -rf bazel
+echo "Done."
+
+################################################################################
+# Prepare the cache.
+################################################################################
+echo $LINE_SEP
+echo "Preparing bazel offline cachedir ..."
+cd ${REPO_TOP}
+# Make bazel forget everything it knows, then download everything.
+${BAZELISK} clean --expunge
+${BAZELISK} fetch --repository_cache=${BAZEL_AIRGAPPED_DIR}/${BAZEL_CACHEDIR} \
+  //... \
+  @bazel_embedded_upstream_toolchain//... \
+  @local_config_cc_toolchains//... \
+  @local_config_platform//... \
+  @local_config_sh//... \
+  @python3_9_toolchains//... \
+  @riscv-compliance//... \
+  @riscv-compliance//... \
+  @rust_darwin_aarch64_toolchains//... \
+  @rust_darwin_x86_64_toolchains//... \
+  @rust_freebsd_x86_64_toolchains//... \
+  @rust_linux_aarch64_toolchains//... \
+  @rust_linux_x86_64_toolchains//... \
+  @rust_windows_x86_64_toolchains//...
```

Comment:
I don't think (hope?) we need to do this for every external repo.

We have to do it for rust because `@rules_rust` creates some weird proxy repo thing that downloads the rust toolchain later.

Reply:
I went one-by-one and added them as the the build errored out saying it was missing each one... so it seems we have to?

To test locally without being in airgapped env I ran:
```
./util/prep-bazel-airgapped-build.sh --force-rebuild && bazel build --distdir=bazel-airgapped/bazel-distdir --repository_cache=bazel-airgapped/bazel-distdir --nofetch //sw/device/silicon_creator/mask_rom
```